site/oauth2-examples-azure.md

```diff
@@ -0,0 +1,237 @@
+
+## Prerequisites to follow this guide
+
+* Have an account in https://portal.azure.com.
+* Docker
+* Openssl
+
+## Register your app
+
+When using **Azure AD as OAuth 2.0 server**, your client app (in our case RabbitMQ) needs a way to trust the security tokens issued to it by the **Microsoft identity platform**. The first step in establishing that trust is by **registering your app** with the identity platform in Azure AD.
+
+<g-emoji class="g-emoji" alias="blue_book" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4d8.png">📘</g-emoji> More details about App registration in Azure AD are available [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app)
+
+Once you have logged onto your account in [Azure Portal](https://portal.azure.com), go to **Azure Active Directory** (use the search bar if you are not able to easily find it).
+
+In the left-hand navigation menu, click on **App Registrations**. Then, select **New registration**.
+
+In the **Register an application** pane, provide the following informations:
+
+* **Name**: the name you would like to give to your application (ex: *rabbitmq-oauth2*)
+* **Supported Account Types**: select **Accounts in this organizational directory only (Default Directory only - Single tenant)** (you can choose other options if you want to enlarge the audience of your app)
+* **Redirect URI**:
+  - On the **Select a platform** drop-down list, select **Single-page application (SPA)**
+  - Configure the **Redirect URI** to: `https://localhost:15671/js/oidc-oauth/login-callback.html`
+
+<g-emoji class="g-emoji" alias="warning" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/26a0.png">⚠️</g-emoji> **IMPORTANT**: As you may have noticed, Azure AD only allows `https` links as **Redirect URI**. To fit this need, we will enable HTTPS for RabbitMQ Management UI, on port `15671`.
+
+Click on **Register**.
+
+![Azure AD OAuth2 App](./img/oauth2/azure-ad-oauth-registered-app.png)
+
+Note the following values, as we will need it later to configure the `rabbitmq_auth_backend_oauth2` on RabbitMQ side:
+* Directory (tenant ID)
+* Application (client) ID
+
+Click on the **Endpoints** tab and, on the right pane that has just opened, copy the value of **OpenID Connect metadata document** (ex: `https://login.microsoftonline.com/{TENANT_ID}/v2.0/.well-known/openid-configuration`) and open it in your browser.
+
+Note the value of the `jwks_uri` key (ex: `https://login.microsoftonline.com/{TENANT_ID}/discovery/v2.0/keys`), as you will also need it later to configure the `rabbitmq_auth_backend_oauth2` on RabbitMQ side
+
+![Azure AD JWKS URI](./img/oauth2/azure-ad-jwks-uri.png)
+
+## Create a secret for your app
+
+Your application needs a **client secret** to prove its identity when requesting a token.
+
+Still on the **App registrations** page, in the left-hand menu, click on **Certificates & Secrets**, then select the **Client secrets** tab.
```

Comment:
What if they are not still on this page, do we need to give the full sequence here on how to get to it...what if they have logged out in the meantime and now need to get back to this page

Reply:
I'd say it is totally fine to remind them on which page they should be now .. by saying "Still on the **App registration** page" ...